src/mp4box/dinf.rs

```diff
@@ -0,0 +1,281 @@
+
+        let mut dref = None;
+
+        let mut current = reader.seek(SeekFrom::Current(0))?;
+        let end = start + size;
+        while current < end {
+            // Get box header.
+            let header = BoxHeader::read(reader)?;
+            let BoxHeader { name, size: s } = header;
+
+            match name {
+                BoxType::DrefBox => {
+                    dref = Some(DrefBox::read_box(reader, s)?);
+                }
+                _ => {
+                    // XXX warn!()
+                    skip_box(reader, s)?;
+                }
+            }
+
+            current = reader.seek(SeekFrom::Current(0))?;
+        }
+
+        if dref.is_none() {
+            return Err(Error::BoxNotFound(BoxType::DrefBox));
+        }
+
+        skip_bytes_to(reader, start + size)?;
+
+        Ok(DinfBox {
+            dref: dref.unwrap(),
+        })
+    }
+}
+
+impl<W: Write> WriteBox<&mut W> for DinfBox {
+    fn write_box(&self, writer: &mut W) -> Result<u64> {
+        let size = self.box_size();
+        BoxHeader::new(self.box_type(), size).write(writer)?;
+        self.dref.write_box(writer)?;
+        Ok(size)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Default, Serialize)]
+pub struct DrefBox {
+    pub version: u8,
+    pub flags: u32,
+    pub url: UrlBox,
+}
+
+impl DrefBox {
+    pub fn get_type(&self) -> BoxType {
+        BoxType::DrefBox
+    }
+
+    pub fn get_size(&self) -> u64 {
+        HEADER_SIZE + HEADER_EXT_SIZE + 4 + self.url.box_size()
+    }
+}
+
+impl Mp4Box for DrefBox {
+    fn box_type(&self) -> BoxType {
+        return self.get_type();
+    }
+
+    fn box_size(&self) -> u64 {
+        return self.get_size();
+    }
+
+    fn to_json(&self) -> Result<String> {
+        Ok(serde_json::to_string(&self).unwrap())
+    }
+
+    fn summary(&self) -> Result<String> {
+        let s = format!("");
+        Ok(s)
+    }
+}
+
+impl<R: Read + Seek> ReadBox<&mut R> for DrefBox {
+    fn read_box(reader: &mut R, size: u64) -> Result<Self> {
+        let start = box_start(reader)?;
+
+        let mut current = reader.seek(SeekFrom::Current(0))?;
+
+        let (version, flags) = read_box_header_ext(reader)?;
+        let end = start + size;
+
+        let mut url = None;
+
+        let entry_count = reader.read_u32::<BigEndian>()?;
+        for _i in 0..entry_count {
+            if current >= end {
+                break;
+            }
+
+            // Get box header.
+            let header = BoxHeader::read(reader)?;
+            let BoxHeader { name, size: s } = header;
+
+            match name {
+                BoxType::UrlBox => {
+                   url = Some(UrlBox::read_box(reader, s)?);
+                }
+                _ => {
+                    skip_box(reader, s)?;
+                }
+            }
+
+            current = reader.seek(SeekFrom::Current(0))?;
+        }
+
+        if url.is_none() {
+            return Err(Error::BoxNotFound(BoxType::UrlBox));
+        }
```

Comment:
Can this be optional? I have a qt file I tried without the url box and got this error.

Reply:
Done!